src/compiler/visitor.ts

```diff
@@ -222,7 +222,7 @@ namespace ts {
             // Names
 
             case SyntaxKind.Identifier:
-                return updateIdentifier(<Identifier>node, nodesVisitor((<Identifier>node).typeArguments, visitor, isTypeNode));
+                return updateIdentifier(<Identifier>node, nodesVisitor((<Identifier>node).typeArguments, visitor, or(isTypeNode, isTypeParameterDeclaration)));

```

Comment:
I understand your reasoning for adding `or`, but I still believe `or(isTypeNode, ...)` should be pulled out into a separate function.